state-sync/aptos-data-client/src/client.rs

```diff
@@ -100,17 +101,21 @@ impl AptosDataClient {
         storage_service_client: StorageServiceClient<NetworkClient<StorageServiceMessage>>,
         runtime: Option<Handle>,
     ) -> (Self, DataSummaryPoller) {
+        // Wrap the configs in an Arc (to be shared across components)
+        let base_config = Arc::new(base_config);
```

Comment:
why not &?

Reply:
Aah, because the structs are cloned during program execution, so `Arc` makes the code simpler and avoids me having to add explicit lifetimes to them (e.g., `'static`).

Reply:
As a c++ engineer in general I dislike Arc  (std::shared_ptr in c++) :) 
It creates unnecessary overhead, unclear ownership, potentially unexpected memory consumption, and destruction at undesirable time.